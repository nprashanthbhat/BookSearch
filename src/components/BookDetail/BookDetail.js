import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import StarRatingComponent from 'react-star-rating-component';

const BookDetail = (props) => {

  const { title, image, author, rating,
    description, ratingsCount, reviewsCount } = props.bookInfo;
  const { imageContainer, imageStyle, detailsContainer,
    titleStyle, descContainer, rowStyle, textStyle } = styles;

    return(
      <table>
        <tr style={rowStyle}>
          <td style={imageContainer}>
            {'image' in props.bookInfo && <div>
              <img src={image ? image : ''} alt='' style={imageStyle}/>
            </div>
            }
            {'rating' in props.bookInfo && <div>
                <StarRatingComponent starCount={5} value={rating} />
              <div>
                {rating} / 5
              </div>
            </div>}
          </td>
          <td style={detailsContainer}>
              <div style={titleStyle}>{title}</div>
            {'author' in props.bookInfo && <div
              style={styles.authorInfo}>{`By `}{author}
            </div>
            }
            <div style={textStyle}>
              {'ratingsCount' in props.bookInfo && <text><b>{ratingsCount}</b> ratings &nbsp;</text>}
              {'reviewsCount' in props.bookInfo && <text><b>{reviewsCount}</b> reviews</text>}
            </div>
            {'description' in props.bookInfo && <div
              style={descContainer}>{ReactHtmlParser(description)}
            </div>
            }
          </td>
        </tr>
      </table>
  );
}

const styles = {
  imageContainer: {
    margin: 5,
    minWidth: '8em',
    textAlign: 'center'
  },
  imageStyle: {
    width: '100%',
    height: '100%'
  },
  detailsContainer: {
    padding: 10
  },
  titleStyle: {
    fontSize: 24,
    fontWeight: 'bold'
  },
  authorInfo: {
    fontSize: '18',
    fontStyle: 'italic',
    paddingBottom: 10
  },
  descContainer: {
    overflow : 'scroll'
  },
  rowStyle: {
    verticalAlign: 'top'
  },
  textStyle: {
    color: 'green',
    fontSize: 15,
    padding: 5
  }
};

export default BookDetail;
