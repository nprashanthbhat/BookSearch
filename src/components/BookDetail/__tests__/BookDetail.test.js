import React from 'react';
import { shallow } from 'enzyme';
import BookDetail from '../BookDetail';

describe('BookDetail', () => {
  const props = {
    bookInfo: []
  };

  it('book details array', () => {
    props.bookInfo = [{
      title: 'The Old Man and the Sea',
      image: 'https://images.gr-assets.com/books/1329189714m/2165.jpg',
      author: 'Ernest Hemingway',
      rating: '3.75',
      description: 'It is the story of an old Cuban fisherman and his supreme ordeal: a relentless, agonizing battle with a giant marlin far out in the Gulf Stream.',
      ratingsCount: '594120',
      reviewsCount: '13850'
    }];
  });

const bookdetails = shallow(<BookDetail {...props} />);

it ('renders without crashing', () => {
  expect(bookdetails).toMatchSnapshot();
});

});
