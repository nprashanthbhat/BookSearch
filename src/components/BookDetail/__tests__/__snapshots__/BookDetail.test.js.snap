// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookDetail
    bookInfo={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <tr
        style={
          Object {
            "verticalAlign": "top",
          }
        }
      >
        <td
          style={
            Object {
              "margin": 5,
              "minWidth": "8em",
              "textAlign": "center",
            }
          }
        />
        <td
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          />
          <div
            style={
              Object {
                "color": "green",
                "fontSize": 15,
                "padding": 5,
              }
            }
          />
        </td>
      </tr>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            style={
              Object {
                "margin": 5,
                "minWidth": "8em",
                "textAlign": "center",
              }
            }
          />,
          <td
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <div
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            />
            <div
              style={
                Object {
                  "color": "green",
                  "fontSize": 15,
                  "padding": 5,
                }
              }
            />
          </td>,
        ],
        "style": Object {
          "verticalAlign": "top",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              false,
            ],
            "style": Object {
              "margin": 5,
              "minWidth": "8em",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "bold",
                  }
                }
              />,
              false,
              <div
                style={
                  Object {
                    "color": "green",
                    "fontSize": 15,
                    "padding": 5,
                  }
                }
              />,
              false,
            ],
            "style": Object {
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "style": Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  false,
                  false,
                ],
                "style": Object {
                  "color": "green",
                  "fontSize": 15,
                  "padding": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                false,
                false,
              ],
              "type": "div",
            },
            false,
          ],
          "type": "td",
        },
      ],
      "type": "tr",
    },
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tr
          style={
            Object {
              "verticalAlign": "top",
            }
          }
        >
          <td
            style={
              Object {
                "margin": 5,
                "minWidth": "8em",
                "textAlign": "center",
              }
            }
          />
          <td
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <div
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            />
            <div
              style={
                Object {
                  "color": "green",
                  "fontSize": 15,
                  "padding": 5,
                }
              }
            />
          </td>
        </tr>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <td
              style={
                Object {
                  "margin": 5,
                  "minWidth": "8em",
                  "textAlign": "center",
                }
              }
            />,
            <td
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <div
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "bold",
                  }
                }
              />
              <div
                style={
                  Object {
                    "color": "green",
                    "fontSize": 15,
                    "padding": 5,
                  }
                }
              />
            </td>,
          ],
          "style": Object {
            "verticalAlign": "top",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                false,
              ],
              "style": Object {
                "margin": 5,
                "minWidth": "8em",
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              false,
              false,
            ],
            "type": "td",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                    }
                  }
                />,
                false,
                <div
                  style={
                    Object {
                      "color": "green",
                      "fontSize": 15,
                      "padding": 5,
                    }
                  }
                />,
                false,
              ],
              "style": Object {
                "padding": 10,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "style": Object {
                    "fontSize": 24,
                    "fontWeight": "bold",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    false,
                    false,
                  ],
                  "style": Object {
                    "color": "green",
                    "fontSize": 15,
                    "padding": 5,
                  },
                },
                "ref": null,
                "rendered": Array [
                  false,
                  false,
                ],
                "type": "div",
              },
              false,
            ],
            "type": "td",
          },
        ],
        "type": "tr",
      },
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
